socketio/src/asynchronous/client/client.rs

```diff
@@ -0,0 +1,733 @@
+        let socket = ClientBuilder::new(url)
+            .transport_type(TransportType::Any)
+            .connect_manual()
+            .await?;
+        test_socketio_socket(socket, "/".to_owned()).await
+    }
+
+    async fn test_socketio_socket(mut socket: Client, nsp: String) -> Result<()> {
+        let _: Option<Packet> = Some(socket.next().await.unwrap()?);
+
+        println!("0");
+        let packet: Option<Packet> = Some(socket.next().await.unwrap()?);
+
+        assert!(packet.is_some());
+
+        let packet = packet.unwrap();
+
+        assert_eq!(
+            packet,
+            Packet::new(
+                PacketId::Event,
+                nsp.clone(),
+                Some("[\"Hello from the message event!\"]".to_owned()),
+                None,
+                0,
+                None,
+            )
+        );
+        println!("1");
+
+        let packet: Option<Packet> = Some(socket.next().await.unwrap()?);
+
+        assert!(packet.is_some());
+
+        let packet = packet.unwrap();
+
+        assert_eq!(
+            packet,
+            Packet::new(
+                PacketId::Event,
+                nsp.clone(),
+                Some("[\"test\",\"Hello from the test event!\"]".to_owned()),
+                None,
+                0,
+                None
+            )
+        );
+        println!("2");
+        let packet: Option<Packet> = Some(socket.next().await.unwrap()?);
+
+        assert!(packet.is_some());
+
+        let packet = packet.unwrap();
+        assert_eq!(
+            packet,
+            Packet::new(
+                PacketId::BinaryEvent,
+                nsp.clone(),
+                None,
+                None,
+                1,
+                Some(vec![Bytes::from_static(&[4, 5, 6])]),
+            )
+        );
+
+        let packet: Option<Packet> = Some(socket.next().await.unwrap()?);
+
+        assert!(packet.is_some());
+
+        let packet = packet.unwrap();
+        assert_eq!(
+            packet,
+            Packet::new(
+                PacketId::BinaryEvent,
+                nsp.clone(),
+                Some("\"test\"".to_owned()),
+                None,
+                1,
+                Some(vec![Bytes::from_static(&[1, 2, 3])]),
+            )
+        );
+
+        let cb = |message: Payload, _| {
+            async {
+                println!("Yehaa! My ack got acked?");
+                if let Payload::String(str) = message {
+                    println!("Received string ack");
+                    println!("Ack data: {}", str);
+                }
+            }
+            .boxed()
+        };
+
+        assert!(socket
+            .emit_with_ack(
+                "test",
+                Payload::String("123".to_owned()),
+                Duration::from_secs(10),
+                cb
+            )
+            .await
+            .is_ok());
+
+        Ok(())
+    }
+
+    // TODO: 0.3.X add secure socketio server
```

Comment:
nit: Is this test a blocker on 0.4.0 coming out of alpha?

Reply:
I don't think so